src/transformers/modeling_prophetnet.py

```diff
@@ -0,0 +1,1461 @@
+            if self.output_hidden_states:
+                all_hidden_states += (x,)
+            layer_state = decoder_cached_states[idx] if decoder_cached_states is not None else None
+            x, layer_self_attn, layer_past = decoder_layer(
+                x,
+                encoder_hidden_states,
+                encoder_attn_mask=encoder_padding_mask,
+                layer_state=layer_state,
+                self_attn_mask=self_attn_mask,
+                output_attentions=output_attentions,
+                ngram_mask_matrix=ngram_mask_matrix,
+                i_buckets_main_stream=i_buckets_main_stream,
+                i_bucket_relative_stream=i_bucket_relative_stream,
+                real_positions=real_positions
+            )
+            if use_cache:
+                next_decoder_cache.append(layer_past.copy())
+            if output_attentions:
+                all_self_attns += (layer_self_attn,)
+        # TODO [(1+ngram)*T, B, C] -> [B, (1+ngram)*T, C]
+        x_list = x.transpose(0, 1).chunk(1 + self.ngram, 1)
+        encoder_hidden_states = encoder_hidden_states.transpose(0, 1)
+        if use_cache:
+            next_cache = ((encoder_hidden_states, encoder_padding_mask), next_decoder_cache)
+        else:
+            next_cache = None
+        return x_list, next_cache, all_hidden_states, list(all_self_attns)
+
+
+
+
+
+
+
+def shift_tokens_right(input_ids, pad_token_id):
+    """Shift input ids one token to the right, and wrap the last non pad token (usually <eos>)."""
+    prev_output_tokens = input_ids.clone()
+    index_of_eos = (input_ids.ne(pad_token_id).sum(dim=1) - 1).unsqueeze(-1)
+    prev_output_tokens[:, 0] = input_ids.gather(1, index_of_eos).squeeze()
+    prev_output_tokens[:, 1:] = input_ids[:, :-1]
+    return prev_output_tokens
+
+def make_padding_mask(input_ids, padding_idx=1):
+    """True for pad tokens"""
+    padding_mask = input_ids.eq(padding_idx)
+    if not padding_mask.any():
+        padding_mask = None
+    return padding_mask
+
+def fill_with_neg_inf(t):
+    """FP16-compatible function that fills a input_ids with -inf."""
+    return t.float().fill_(float("-inf")).type_as(t)
+
+def _filter_out_falsey_values(tup) -> Tuple:
+    """Remove entries that are None or [] from an iterable."""
+    return tuple(x for x in tup if isinstance(x, torch.Tensor) or x)
+
+@add_start_docstrings(
+    "The bare ProphetNet Model transformer outputting raw hidden-states without any specific head on top.",
+    PROPHETNET_START_DOCSTRING,
+    PROPHETNET_INPUTS_DOCSTRING,
+)
+class ProphetNetModel(ProphetNetPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        padding_idx, vocab_size, dim_size = config.pad_token_id, config.vocab_size, config.d_model
+        self.embed_tokens = nn.Embedding(vocab_size, dim_size, padding_idx=padding_idx)
+        nn.init.normal_(self.embed_tokens.weight, mean=0, std=dim_size ** -0.5)
```

Comment:
what is this doing ? -> normally we handle the `init` in the `ProphetNetPreTrainedModel`, such as here: https://github.com/huggingface/transformers/blob/31da35cc8939e19a292cb7f15cad5c9a1ddf7f23/src/transformers/modeling_bert.py#L575